src/java/org/apache/cassandra/db/compression/CompressionDictionaryDetailsTabularData.java

```diff
@@ -0,0 +1,229 @@
+    {
+        String keyspace = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KEYSPACE_NAME);
+        String table = (String) compositeData.get(CompressionDictionaryDetailsTabularData.TABLE_NAME);
+        long dictId = (Long) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_ID_NAME);
+        byte[] dictionaryBytes = (byte[]) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_NAME);
+        String kind = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KIND_NAME);
+        int checksum = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.CHECKSUM_NAME);
+        int size = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.SIZE_NAME);
+
+        CompressionDictionaryPojo pojo = new CompressionDictionaryPojo();
+        pojo.keyspace = keyspace;
+        pojo.table = table;
+        pojo.dictId = dictId;
+        pojo.dict = dictionaryBytes;
+        pojo.kind = kind;
+        pojo.dictChecksum = checksum;
+        pojo.dictLength = size;
+
+        pojo.validate();
+
+        return pojo;
+    }
+
+    public static class CompressionDictionaryPojo
+    {
+        public String keyspace;
+        public String table;
+        public long dictId;
+        public byte[] dict;
+        public String kind;
+        public int dictChecksum;
+        public int dictLength;
+
+        /**
+         * Dictionary is valid if, keyspace and table are specified, dictionary id is strictly positive integer,
+         * dictionary byte array is not nor not empty,
+         * kind corresponds to {@code Kind}, checksum and size are bigger than 0.
+         */
+        public void validate()
+        {
+            if (keyspace == null)
+                throw new IllegalArgumentException("keyspace not specified");
+            if (table == null)
+                throw new IllegalArgumentException("table not specified");
+            if (dictId <= 0)
+                throw new IllegalArgumentException("Provided dictionary id is lower than 0, it is '" + dictId + "'.'");
+            if (dict == null || dict.length == 0)
+                throw new IllegalArgumentException("Provided dictionary byte array is null or empty.");
+            if (kind == null)
+                throw new IllegalArgumentException("Provided kind is null.");
+
+            CompressionDictionary.Kind dictionarykind;
+
+            try
+            {
+                dictionarykind = CompressionDictionary.Kind.valueOf(kind);
+            }
+            catch (IllegalArgumentException ex)
+            {
+                throw new IllegalArgumentException("There is no such dictionary kind like '" + kind + "'. Available kinds: " + Arrays.asList(CompressionDictionary.Kind.values()));
+            }
+
+            if (dictChecksum <= 0)
+                throw new IllegalArgumentException("Checksum has to be strictly positive number, it is '" + dictChecksum + "'.");
```

Comment:
I think the checksum could be negative. The checksum is from crc32, which produce 32 bits. In signed integer, half of the values are negative.

Reply:
aha interesting, good to know. I have never seen it negative so far.